scene/gui/file_dialog.cpp

```diff
@@ -790,6 +812,23 @@ void FileDialog::_filter_selected(int) {
 	update_file_list();
 }
 
+void FileDialog::_filename_filter_changed() {
+	update_filename_filter();
+	TreeItem *item = tree->get_item_at_position(Vector2(0, 0));
+	if (item) {
+		tree->call_deferred("set_selected", item, 0);
+	}
+	update_file_list();
+}
```

Comment:
```suggestion
	update_filename_filter();
	update_file_list();
	callable_mp(this, &FileDialog::_tree_select_first).call_deferred();
}

void FileDialog::_tree_select_first() {
	if (tree->get_root() && tree->get_root()->get_first_child()) {
		tree->get_root()->get_first_child()->select(0);
	}
}
```

`update_file_list()` is regenerating tree and old item won't exist when deferred call is executed (and will likely crash), instead you can add a new function call it deferred and get first item inside of it.